### PR TITLE
bugfix: using rewriter to erase an op inside a pattern rewriter

### PR DESCRIPTION
**Pull request by wyzero**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc [hunk 1]
```diff
@@ -175,7 +175,7 @@ LogicalResult InputInlineFusionPattern::processParallelOp(
     // Clean all the ops that do not have LoadOps inside the nested
     // ParallelOps and is not the ancestor of any ops that have LoadOps
     // inside the nested ParallelOps.
-    cleanUnusedLhloOps(parent_block);
+    cleanUnusedLhloOps(parent_block, &rewriter);
```
Comment (fortianyou) [inline tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc:178]: Are there other usages of cleanUnusedLhloOps elsewhere?

Comment (fortianyou) [reply tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc:178]: image

There is another cleanUnusedLhloOps definition will it introduce problems too?

**Review by fortianyou — APPROVED** [review]
LGTM

**Review by qiuxiafei — APPROVED** [review]

File: tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc:178]: > Are there other usages of cleanUnusedLhloOps elsewhere?

These are not inside a rewrite pattern.

Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc:178]: > image
> 
> There is another cleanUnusedLhloOps definition will it introduce problems too?

seems that it is a duplicate implementation. I'll remove it in the next PR.

Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/input_inline_fusion_pattern.cc:178]: I double-check the code: it's inside tf_community, not inside disc. I think it's the code we upstream to the community in the last year.

**Pull request merged**